File: frontend/src/types/milestone.ts
export interface Milestone {
  id: number;
  project_id: string;
  title: string;
  milestone_type: string | null;
  due_date: string | null;
  completed_date: string | null;
  status: string | null;
  owner: string | null;
  jira_epic_key: string | null;
  progress_pct: number;
  sort_order: number | null;
  notes: string | null;
  created_at: string | null;
  updated_at: string | null;
}

File: backend/app/schemas/milestone.py
from typing import Optional

from pydantic import BaseModel


class MilestoneOut(BaseModel):
    id: int
    project_id: str
    title: str
    milestone_type: Optional[str] = None
    due_date: Optional[str] = None
    completed_date: Optional[str] = None
    status: Optional[str] = None
    owner: Optional[str] = None
    jira_epic_key: Optional[str] = None
    progress_pct: float = 0.0
    sort_order: Optional[int] = None
    notes: Optional[str] = None
    created_at: Optional[str] = None
    updated_at: Optional[str] = None
